src/engine.rs

```diff
@@ -501,9 +526,9 @@ impl evm::backend::Backend for Engine {
         Engine::get_code(&address)
     }
 
-    /// Get storage value of address at index.
-    fn storage(&self, address: Address, index: H256) -> H256 {
-        Engine::get_storage(&address, &index)
+    /// Get storage value of address at key.
+    fn storage(&self, address: Address, key: H256) -> H256 {
+        Engine::get_storage(&address, &key, self.generation)
```

Comment:
I think this is wrong. If a user account calls a contract the generation should be that of the contract, not the user account. As is, it looks like `self.generation` is set by the user account initiating the transaction.

Even if you were to change `self.generation` to match the contract that was being called by the user (if any), then it still wouldn't work because that contract could then call out to a different contract, and then its generation should be used.

What I am getting at is that `generation` is not a constant for the whole transaction, it scoped to a particular call, which means it cannot come from a field of `self` (since SputnikVM manages the call stack, not us). I think the only way to avoid multiple look ups is with caching (or some invasive change to SputnikVM).

Reply:
So, reverting the store generation on creation, all good?

Reply:
Reverted

Reply:
@joshuajbouw you didn't revert the right commit.

Reply:
Oh, wow. What in the world.